src/Agent/NewRelic/Agent/Core/DataTransport/ConnectionManager.cs

```diff
@@ -135,6 +135,8 @@ private void Disconnect()
 
         private void Reconnect()
         {
+            EventBus<StopHarvestEvent>.Publish(new StopHarvestEvent());
```

Comment:
Would it make more sense to have this in Disconnect? That way it would also stop harvesting during shutdown. Or maybe that already happens?

Reply:
> Would it make more sense to have this in Disconnect? That way it would also stop harvesting during shutdown. Or maybe that already happens?

After putting some thought, I think I still like have this as where it is. The reasons are:
1. I would like this to happen as soon as the agent receives a reconnect event. Currently the Disconnect call is wrapped inside a lock which potentially delays the call.
2. The PreCleanShutdownEvent event already calls `_scheduler.StopExecuting(Harvest, TimeSpan.FromSeconds(2));` and it happens before the CleanShutdownEvent, so it would not make any different to make such change.